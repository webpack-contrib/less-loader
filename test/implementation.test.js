import {
  compile,
  getCodeFromBundle,
  getCodeFromLess,
  getCompiler,
  getErrors,
  getWarnings,
} from './helpers';

describe('implementation options', () => {
  it('should work', async () => {
    const testId = './basic.less';
    const compiler = getCompiler(testId, {
      // eslint-disable-next-line global-require
      implementation: require('less'),
    });
    const stats = await compile(compiler);
    const codeFromBundle = getCodeFromBundle(stats, compiler);
    const codeFromLess = await getCodeFromLess(testId);

    expect(codeFromBundle.css).toBe(codeFromLess.css);
    expect(codeFromBundle.css).toMatchSnapshot('css');
    expect(getWarnings(stats)).toMatchSnapshot('warnings');
    expect(getErrors(stats)).toMatchSnapshot('errors');
  });
});
