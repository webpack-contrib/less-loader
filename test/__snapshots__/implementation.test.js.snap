// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`implementation options should work: css 1`] = `
".box {
  color: #fe33ac;
  border-color: #fdcdea;
  background: url(box.png);
}
.box div {
  -webkit-box-shadow: 0 0 5px rgba(0, 0, 0, 0.3);
  box-shadow: 0 0 5px rgba(0, 0, 0, 0.3);
}
"
`;

exports[`implementation options should work: errors 1`] = `Array []`;

exports[`implementation options should work: warnings 1`] = `Array []`;
